smdebug/tensorflow/keras.py

```diff
@@ -479,7 +496,9 @@ def _on_any_batch_begin(self, batch, mode, logs=None):
             self._prepare_collections()
 
         if self._prepared_tensors[mode] is False:
-            if self._validate_exec_function(self._get_exec_function(mode)):
+            if (is_tf_version_2x() and tf.executing_eagerly()) or self._validate_exec_function(
```

Comment:
else?

Reply:
else we don't prepare layers in this call of batch begin